components/dns/main.tf

```diff
@@ -0,0 +1,24 @@
+data "local_file" "configuration" {
+  filename = "${path.cwd}/../../apps.yaml"
+}
+
+locals {
+  apps = yamldecode(data.local_file.configuration.content).apps
+  cname_records = flatten([
+    for app in local.apps : {
+      name      = split(".", replace("${app.externalUrl}", "https://", ""))[0],
+      zone_name = join(".", slice(split(".", "${app.externalUrl}"), 1, length(split(".", "${app.externalUrl}"))))
```

Comment:
can you explain how this works? I don't see how you would be able to determine a zone name without extracting it to another variable really.

what if you had www.bar.hmcts.net. would this find hmcts.net as the zone or bar as the zone?

It might be good enough for now and we can adjust later if needed

Reply:
App proxy is not allowing special chars in domain name. we cannot add `www.bar` or `bar-test`, so we might be good. Added a comment anyway